tests/ssg_test_suite/common.py

```diff
@@ -220,5 +261,31 @@ def send_scripts(domain_ip):
             msg = "Cannot extract data tarball {0}.".format(remote_archive_file)
             logging.error(msg)
             raise RuntimeError(msg)
-
+    os.unlink(archive_file)
     return remote_dir
+
+
+def iterate_over_rules():
+    """Iterate over rule directories which have test scnearios".
+
+    Returns:
+        Named tuple Rule having these fields:
+            directory -- absolute path to the rule "tests" subdirectory
+                         containing the test scenarios in Bash
+            id -- full rule id as it is present in datastream
+            short_id -- short rule ID, the same as basename of the
+            files -- list of executable .sh files in the
+            directory containing the test scenarios in Bash
+    """
+    for dirpath, dirnames, filenames in os.walk(_BENCHMARK_DIR):
+        if "rule.yml" in filenames and "tests" in dirnames:
+            short_rule_id = os.path.basename(dirpath)
+            tests_dir = os.path.join(dirpath, "tests")
+            tests_dir_files = os.listdir(tests_dir)
+            # Filter out everything except the shell test scenarios.
+            # Other files in rule directories are editor swap files
+            # or other content than a test case.
+            scripts = filter(lambda x: x.endswith(".sh"), tests_dir_files)
+            full_rule_id = OSCAP_RULE + short_rule_id
+            result = Rule(tests_dir, full_rule_id, short_rule_id, scripts)
```

Comment:
Please use the kwargs here, so it is obvious that the `result.id == full_rule_id` and the assignment is robust wrt to the `Rule` type possible evolution.